src/nvim/drawline.c

```diff
@@ -1185,21 +1277,48 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
               }
               p_extra = extra;
               c_extra = NUL;
+            } else if (*wp->w_p_stc != NUL && *statuscol.final != ' ') {
+              // empty space plus final character in 'statuscolumn' after wrapping
```

Comment:
A more generic solution (without having this special case) would be to expose a variable indicating if it's a wrapped line. And whoever's setting the `statuscolumn` option can use that if they want.

Reply:
Then we need a way to calculate the number of spaces (or whatever), to put before the desired separator, else the border character would be misaligned.

Reply:
I don't think `%r` is for right align. But there's `%=`, which should push the everything on both sides. So `%=x` should be right aligned `x`.

Reply:
That might be more robust. Would leave you with something like `:let &stc='%{v:redraw_wrap?"":v:redraw_lnum}%=│ '`.

Reply:
Just curious, why was the `redraw_` prefix used? 🤔

Reply:
To group the variables, and because they are only relevant while redrawing a line. Did you have a different prefix in mind?

Reply:
Nah, was just curious. Because the other `v:` variables are not prefixed like this, even though many of them are only valid in some specific context (much like the variables added here which are only valid when `statuscolumn` is being evaluated).